### DISCUSSION TITLE
Question if `prefer-const` works as intended

### DISCUSSION THREAD

**Discussion by UndefinedCpp** [eslint/eslint#16786] [Q&A] [answered]
So basically I have a piece of code like this (TypeScript):
```ts
let dataset: Record<string, number> = {};
input.forEach(item => dataset[item.name] = doSomething(item));
```
With option `prefer-const`, eslint suggests me changing `let` into `const`, because "'dataset' is never reassigned." But I feel like this is somewhat strange, because `const` literally means "constant" - something never change. And in the example, the `dataset` does change, thus (in my opinion) `let` is more readable. Of course, changing `let` into `const` doesn't break the code.

So is it actually a desired behaviour? I use eslint ^8.26.0

**Comment by aladdin-add** [comment, marked as answer]
it's working as expected. using `const` meaning the `dataset` cannot be reassigned, but you can always change its properties.

you can read more in here: https://leanpub.com/understandinges6/read#leanpub-auto-block-level-declarations.
